### DISCUSSION TITLE
Does MACE satisfy E(3) equivariance or just SE(3) equivariance?

### DISCUSSION THREAD

**Discussion by QuantumLab-ZY** [ACEsuit/mace#470] [Q&A] [answered]
Dear Developers,

Hello! After delving into your insightful article and carefully reading the accompanying code of MACE, I have a query that I would greatly appreciate your clarification on. I noticed that in the higher-order tensor products described in your article, there is no explicit consideration for parity (or inversion) equivariance – specifically, parity indices are not included in the provided formulas. Therefore, I am wondering whether MACE is designed to support equivariance under the E(3) group or the more specialized SE(3) group. 

I am looking forward to your valuable insights, which I am confident will significantly enhance my understanding. Thank you in advance for your time and assistance.

Best regards,
Yang Zhong

**Comment by ilyes319** [comment, marked as answer]
Hi @QuantumLab-ZY,

The Default MACE code is E(3) equivariant but does not consider all irreps of the group, but just the one of alternating parity so for example "0e + 1o + 2e". You can view that as a sparsification of the full E(3) group. We observed that this gives the best trade off between accuracy and speed. You can easily consider the full group, either using mace-jax that supports it natively (https://github.com/ACEsuit/mace-jax) or looking at the following branch in mace-torch (https://github.com/ACEsuit/mace/tree/parity-mix-channels).

**Reply by QuantumLab-ZY** [reply]
Thank you for your prompt reply. I now understand that MACE is E(3) equivariant.
